File: Frontend/src/models/AddCouponRequest.ts
class AddCouponRequest {
    companyId?: number;
    category?: string;
    title: string;
    description: string;
    startDate: Date;
    endDate: Date;
    amount: number;
    price: number;
    imageUrl: string;


    constructor(companyId: number, category: string, title: string, description: string, startDate: Date, endDate: Date, amount: number, price: number, imageUrl: string) {
        this.companyId = companyId;
        this.category = category;
        this.title = title;
        this.description = description;
        this.startDate = startDate;
        this.endDate = endDate;
        this.amount = amount;
        this.price = price;
        this.imageUrl = imageUrl;
    }
}

export default AddCouponRequest;

File: Backend/src/main/java/com/ap/requestmodels/AddCouponRequest.java
package com.ap.requestmodels;

import lombok.Data;

import java.math.BigDecimal;
import java.sql.Date;

@Data
public class AddCouponRequest {

    private String title;
    private long companyId;
    private String description;
    private int amount;
    private String category;
    private Date startDate;
    private Date endDate;
    private BigDecimal price;
    private String img;

}
